### PR TITLE
feat(oauth impersonation): add create and delete endpoints

### LINKED ISSUE #1314: Support impersonation OAuth token endpoints
Currently, this library does support the [Create an impersonation OAuth token](https://developer.github.com/enterprise/2.17/v3/enterprise-admin/users/#create-an-impersonation-oauth-token) or [Delete an impersonation OAuth token](https://developer.github.com/enterprise/2.17/v3/enterprise-admin/users/#delete-an-impersonation-oauth-token). 

I'm going to add these endpoints within the next couple of days.

### PR DESCRIPTION
**Pull request by JordanSussman**
closes #1314

### CODE REVIEW

File: github/admin_users.go [hunk 1]
```diff
@@ -59,3 +60,74 @@ func (s *AdminService) DeleteUser(ctx context.Context, username string) (*Respon
 
 	return resp, nil
 }
+
+// ImpersonateUserOptions represents the scoping for the OAuth token.
+type ImpersonateUserOptions struct {
+	Scopes *[]string `json:"scopes,omitempty"`
```
Comment (gmlewis) [inline github/admin_users.go]: Throughout this PR, please change all instances of `*[]string` to `[]string` as the slice is already a reference type.

File: github/admin_users.go [hunk 2]
```diff
@@ -59,3 +60,74 @@ func (s *AdminService) DeleteUser(ctx context.Context, username string) (*Respon
...
+
+func (s OAuthAPP) String() string {
+	return Stringify(s)
+}
+
+// UserAuthorization represents the impersonation response.
+type UserAuthorization struct {
+	ID             *int64     `json:"id,omitempty"`
+	URL            *string    `json:"url,omitempty"`
+	Scopes         *[]string  `json:"scopes,omitempty"`
```
Comment (gmlewis) [inline github/admin_users.go]: Here too.

**Comment by JordanSussman** [comment]
@gmlewis  - Just pushed the requested feedback. Thanks!

**Comment by gmlewis** [comment]
Thank you, @wesleimp!
Merging.

**Pull request merged**